starfish/pipeline/features/spots/detector/__init__.py

```diff
@@ -43,6 +59,11 @@ def _cli(cls, args, print_help=False):
 
         print('Detecting Spots ...')
         image_stack = ImageStack.from_path_or_url(args.input)
+        if args.blobs_stack is not None:
+            blobs_stack = ImageStack.from_path_or_url(args.blobs_stack)  # type: ignore
+            blobs_image = blobs_stack.max_proj(Indices.ROUND, Indices.CH)
+        else:
+            blobs_image = None
         instance = args.spot_finder_algorithm_class(**vars(args))
-        intensities = instance.find(image_stack)
+        intensities = instance.find(image_stack, blobs_image=blobs_image)
```

Comment:
philosophical discussion:

`reference_image_from_max_projection` is passed into the constructor (if I'm reading the code correctly) but `blobs_image` is passed into the find method.  I think this is correct, because `reference_image_from_max_projection` is applied for an entire experiment, but `blobs_image` is different per FOV.

however, i want to make sure everyone is still on board with this distinction.

Reply:
I messed this up in `__init__`. I think reference_image_from_max_projection _probably is_ applied to the whole experiment, but I think it should logically go in the find() method since it shares functionality with `blobs_image`. Technically there is no limitation requiring one to have a `blobs_image` for each FOV, although it would be an odd experimental design that would produce a dataset with only some of the FOVs having this data. 

I've added `reference_image_from_max_projection` to the CLI.